File: apps/web/src/lib/catalog-title.ts
export function mediaTypeLabel(type: string): string {
  const normalized = type.trim().toLowerCase()
  if (normalized === "movie") return "Movie"
  if (normalized === "series") return "Series"
  return normalized ? normalized.replace(/^[a-z]/, (letter) => letter.toUpperCase()) : "Catalog"
}

export function formatCatalogTitle(title: string, type: string): string {
  const trimmed = title.trim()
  const label = mediaTypeLabel(type)
  if (!trimmed || trimmed.toLowerCase() === label.toLowerCase()) return label
  if (trimmed.toLowerCase().endsWith(` - ${label.toLowerCase()}`)) return trimmed
  return `${trimmed} - ${label}`
}

File: apps/mobile/composeApp/src/commonMain/kotlin/media/conduit/mobile/account/CatalogTitle.kt
package media.conduit.mobile.account

fun mediaTypeLabel(type: String): String {
    val normalized = type.trim().lowercase()
    return when (normalized) {
        "movie" -> "Movie"
        "series" -> "Series"
        else -> normalized.replaceFirstChar { it.uppercase() }.ifBlank { "Catalog" }
    }
}

fun formatCatalogTitle(title: String, type: String): String {
    val trimmed = title.trim()
    val label = mediaTypeLabel(type)
    if (trimmed.isBlank() || trimmed.equals(label, ignoreCase = true)) return label
    if (trimmed.endsWith(" - $label", ignoreCase = true)) return trimmed
    return "$trimmed - $label"
}
